features/fields_extensie.feature

```diff
@@ -1,126 +1,544 @@
+      | title    | Een of meerdere parameters zijn niet correct.                                                               |
+      | status   | 400                                                                                                         |
+      | detail   | De foutieve parameter(s) zijn: fields.                                                                      |
+      | code     | paramsValidation                                                                                            |
+      | instance | /personen                                                                                                   |
+      En bevat de response een invalidParams met de volgende gegevens
+      | code     | name          | reason                                                |
+      | required | fields        | Parameter bevat een porpertynaam met onjuiste casing. |
+
+    Scenario: opvragen veld met fields door opgeven naam veld die meerdere keren in de resource voorkomt
+      Gegeven het systeem heeft een persoon met de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999994086 |
+      | geslachtsaanduiding | M         |
+      En de persoon heeft de volgende geboorte gegevens
+      | naam                                      | waarde   |
+      | datum (01.03.10)                          | 19860401 |
+      | land.code                                 | 6030     |
+      | plaats.code                               | 0518     |
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                                   |
+      | type                | RaadpleegMetBurgerservicenummer          |
+      | burgerservicenummer | 999994086                                |
+      | fields              | code                                     |
+      Dan bevat de response de volgende gegevens
+      | naam     | waarde                                                                                                      |
+      | type     | https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?#System_Net_HttpStatusCode_BadRequest |
+      | title    | Een of meerdere parameters zijn niet correct.                                                               |
+      | status   | 400                                                                                                         |
+      | detail   | De foutieve parameter(s) zijn: fields.                                                                      |
+      | code     | paramsValidation                                                                                            |
+      | instance | /personen                                                                                                   |
+      En bevat de response een invalidParams[0] met de volgende gegevens
```

Comment:
de foutafhandeling voor deze situatie is nu nog niet gestandaardiseerd.
In BRK krijg ik dan bijvoorbeeld:
```
        {
            "name": "fields",
            "code": "fields",
            "reason": "Deel van de parameterwaarde niet uniek: waarde. Er zijn meerdere resultaten gevonden voor waarde: aardcultuurbebouwd.waarde, aardcultuuronbebouwd.waarde, +6 andere."
        }
```

In BRP implementatie in Den Haag krijg ik dan bijvoorbeeld:
```
        {
            "name": "fields",
            "code": "fields",
            "reason": "Deel van de parameterwaarde niet correct: code."
        }
```

Ik persoonlijk vind de reason zoals die in de BRK API zit het duidelijkst

Reply:
Mee eens. Ik heb het nu op de common gebaseerd. 
Is het afdoende als we deze reason in deze feature opnemen ? Dat wijkt dan enigszins af van de foutafhandeling feature in common, dat kunnen we natuurlijk beschrijven in de inleiding, Dat betekent ook dat we de foutafhandeling uit common als een soort basis zien waarop uitgebreid kan worden naar behoefte.

Reply:
> Dat betekent ook dat we de foutafhandeling uit common als een soort basis zien waarop uitgebreid kan worden naar behoefte.

Dat lijkt mij exact de bedoeling

Reply:
Aangepast